packages/scene-composer/src/hooks/useRuleResult.ts

```diff
@@ -1,13 +1,18 @@
 import { useMemo } from 'react';
 import { Primitive } from '@iot-app-kit/core';
 
-import { IValueDataBinding } from '../interfaces';
+import { IValueDataBinding, TargetMetadata } from '../interfaces';
 import { useSceneDocument, useDataStore } from '../store';
 import { useSceneComposerId } from '../common/sceneComposerIdContext';
 import { dataBindingValuesProvider, ruleEvaluator } from '../utils/dataBindingUtils';
 
 import useBindingData from './useBindingData';
 
+export interface RuleReturnResult {
+  target: string | number;
+  targetMetadata?: TargetMetadata<string>;
+}
```

Comment:
this type should be defined in dataBindingUtils file because `useRuleResult` imports from `dataBindingUtils`. if you add it here, `dataBindingUtils` imports from `useRuleResult` and creates the circular dependency

Reply:
Yeah I wanted to know what is the best place to declare this type.